controllers/factory/vmagent_test.go

```diff
@@ -713,7 +735,7 @@ func TestBuildRemoteWrites(t *testing.T) {
 			name: "test oauth2",
```

Comment:
This test case is pretty similar with the one handled on this pull request: Using remote-write secret without basic auth. (In this case, OAuth2 is being used.)

At first, I wonder why this test case works, but I found out that secret is passed from `ssCache`, not from the `loadScrapeSecrets` in **scrapes.go**.  In the normal operation, secrets are passed from `loadScrapeSecrets`, and without this patch, `loadScrapeSecrets` won't read the secret if `basicAuth` isn't set.

Reply:
Thanks for pointing, will take a look on it.